### PR TITLE
add router hooks

### PR DESCRIPTION
**Pull request by jackbrown**
This is to give us good compatibility with existing routers and make adoption easier :)

### CODE REVIEW

File: src/controllers/hooks/router-store/index.tsx [hunk 1]
```diff
@@ -18,3 +26,32 @@ export const useRouterActions = (): RouterActionsType => {
...
   return allActions;
 };
+
+export const useLocation = (): Location => {
+  const [routerState] = useRouterStore();
+
+  return routerState.location;
+};
+
+export const useHistory = (): BrowserHistory => {
```
Comment (albertogasparin) [inline src/controllers/hooks/router-store/index.tsx:36]: We always said that passing history around was a bad idea. Are we changing idea now?

File: src/controllers/hooks/router-store/index.tsx [hunk 2]
```diff
@@ -18,3 +26,32 @@ export const useRouterActions = (): RouterActionsType => {
...
+  const [routerState] = useRouterStore();
+
+  return routerState.history;
+};
+
+export const useParams = (): MatchParams => {
```
Comment (albertogasparin) [inline src/controllers/hooks/router-store/index.tsx:42]: If we expose something like this, let's make it more powerful: allow a white list of params to look after, so consumers re-render only when one of those params change. Also, should we expose some actions with it to update them and change the url?

Moreover, I would do the same thing for `useQueryParams()` so we can get that for query parameters. And would be cool to have actions to add/remove/update the query string too

File: src/controllers/hooks/router-store/index.tsx [hunk 3]
```diff
@@ -18,3 +26,32 @@ export const useRouterActions = (): RouterActionsType => {
...
+  const [routerState] = useRouterStore();
+
+  return routerState.match.params;
+};
+
+export const useRouteMatch = (route?: string | Route): Match => {
```
Comment (albertogasparin) [inline src/controllers/hooks/router-store/index.tsx:48]: I'm not entirely sure we should expose this. Do we have an actual use case?

File: src/controllers/hooks/router-store/index.tsx (hunk 1 above)
Comment (rohan-deshpande) [reply src/controllers/hooks/router-store/index.tsx:36]: Yeah I agree here, I think we should be careful about exposing it

File: src/controllers/hooks/router-store/index.tsx (hunk 3 above)
Comment (rohan-deshpande) [reply src/controllers/hooks/router-store/index.tsx:48]: What is the difference here between this and `useRouter`? Can't we get the current match from there?

Comment (jackbrown) [reply src/controllers/hooks/router-store/index.tsx:48]: there's no difference really, this is more of a compatibility thing. I'll remove this one and `useHistory` as we don't really need to support them.

**Pull request closed without merging**